src/interface/paint_path.c

```diff
@@ -0,0 +1,775 @@
+    // ebp
+    // esp: [ esi, ???, 000]
+
+    // Probably drawing benches etc.
+
+    rct_drawpixelinfo * dpi = RCT2_GLOBAL(0x140E9A8, rct_drawpixelinfo*);
+
+    if (dpi->zoom_level <= 1) {
+        if (!(RCT2_GLOBAL(0x9DEA6F, uint8) & 1)) {
+            uint8 additions = map_element->properties.path.additions & 0xF;
+            if (additions != 0) {
+                RCT2_GLOBAL(RCT2_ADDRESS_PAINT_SETUP_CURRENT_TYPE, uint8) = VIEWPORT_INTERACTION_ITEM_FOOTPATH_ITEM;
+                if (dword_F3EF74 != 0) {
+                    RCT2_GLOBAL(RCT2_ADDRESS_PAINT_SETUP_CURRENT_TYPE, uint8) = VIEWPORT_INTERACTION_ITEM_NONE;
+                }
+
+                // loc_6A3F9A: No idea how what is happening here
+
+                RCT2_GLOBAL(RCT2_ADDRESS_PAINT_SETUP_CURRENT_TYPE, uint8) = VIEWPORT_INTERACTION_ITEM_FOOTPATH;
+
+                if (dword_F3EF74 != 0) {
+                    RCT2_GLOBAL(RCT2_ADDRESS_PAINT_SETUP_CURRENT_TYPE, uint8) = VIEWPORT_INTERACTION_ITEM_NONE;
+                }
+            }
+        }
+
+        // Redundant zoom-level check removed
+        RCT2_GLOBAL(0xF3EF78, uint32) = dword_F3EF6C->image | dword_F3EF70;
+        //RCT2_CALLPROC_X(0x6A4101, 0, 0, 0, 0, (int) map_element, 0, 0);
+        sub_6A4101(map_element, height, bp, word_F3F038, dword_F3EF6C, dword_F3EF6C->image | dword_F3EF70);
+    }
+
+    // This is about tunnel drawing
+    uint8 direction = (map_element->properties.path.type + get_current_rotation()) & 0x03;
+    uint8 diagonal = map_element->properties.path.type & 0x04;
+    uint8 bl = direction | diagonal;
+
+    if (bp & 2) {
+        // Bottom right of tile is a tunnel
+        registers regs = {0};
+        regs.eax = 0xFFFFFFFF;
+        if (bl == 5) {
+            regs.ax = (height + 16) / 16;
+            regs.ah = 0x0A;
+        } else if (bp & 1) {
+            regs.ax = height / 16;
+            regs.ah = 0x0B;
+        } else {
+            regs.ax = height / 16;
+            regs.ah = 0x0A;
+
+        }
+        RCT2_ADDRESS(0x009E30B6, uint32)[RCT2_GLOBAL(0x141F56B, uint8) / 2] = regs.eax;
+        RCT2_GLOBAL(0x141F56B, uint8)++;
+    }
+
+    if (!(bp & 4)) {
+        return;
+    }
+
+    // Bottom left of the tile is a tunnel
+    registers regs = {0};
+    regs.eax = 0xFFFFFFFF;
+    if (bl == 6) {
+        regs.ax = (height + 16) / 16;
+        regs.ah = 0x0A;
+    } else if (bp & 8) {
+        regs.ax = height / 16;
+        regs.ah = 0x0B;
+    } else {
+        regs.ax = height / 16;
+        regs.ah = 0x0A;
+
+    }
+    RCT2_ADDRESS(0x009E3138, uint32)[RCT2_GLOBAL(0x141F56A, uint8) / 2] = regs.eax;
+    RCT2_GLOBAL(0x141F56A, uint8)++;
+}
+
+void viewport_path_paint_setup(uint8 direction, int height, rct_map_element * map_element)
+{
+    RCT2_GLOBAL(RCT2_ADDRESS_PAINT_SETUP_CURRENT_TYPE, uint8) = VIEWPORT_INTERACTION_ITEM_FOOTPATH;
+
+    uint32 ecx = get_current_rotation();
+    bool word_F3F038 = false;
+
+    uint32 dword_F3EF74 = 0;
+    uint32 dword_F3EF70 = 0;
```

Comment:
F3EF70 is the path image flags (and colour) and F3EF74 is the path bits (i.e. benches) image flags (and colour)